pkg/kubelet/pod_workers.go

```diff
@@ -626,6 +632,16 @@ func (p *podWorkers) UpdatePod(options UpdatePodOptions) {
 		// the channel is empty, so buffer of size 1 is enough.
 		podUpdates = make(chan podWork, 1)
 		p.podUpdates[uid] = podUpdates
+		fullname := kubecontainer.GetPodFullName(pod)
```

Comment:
This could belong inside isAllowedToStart

Reply:
This whole chunk only applies for static pods (which means inside isAllowedToStart it's after the isStaticPod check)